### PR TITLE
Fixed the xenoarch suspension generator rotate verb

### PR DESCRIPTION
**Pull request by DamianX**
Verbs make me sad

### CODE REVIEW

**Review by PJB3005 — APPROVED** [review]

File: code/modules/research/xenoarchaeology/tools/suspension_generator.dm [hunk 1]
```diff
@@ -326,9 +326,19 @@
...
 
-/obj/machinery/suspension_gen/verb/toggle()
+/obj/machinery/suspension_gen/verb/rotate_cw()
 	set src in view(1)
-	set name = "Rotate suspension gen (clockwise)"
+	set name = "Rotate suspension gen (Clockwise)"
+	set category = "Object"
+
+	if(anchored)
+		to_chat(usr, "<span class='warning'>You cannot rotate [src], it has been firmly fixed to the floor.</span>")
```
Comment (ghost) [inline code/modules/research/xenoarchaeology/tools/suspension_generator.dm:335]: Not \the [src]?

**Comment by Kurfursten** [comment]
If verbs make you sad, why not add a couple rotate buttons to the interact menu?

File: code/modules/research/xenoarchaeology/tools/suspension_generator.dm (hunk 1 above)
Comment (Exxion) [reply code/modules/research/xenoarchaeology/tools/suspension_generator.dm:335]: We should probably have a style guide

**Comment by DamianX** [comment]
That'd be pretty nice. How would that work? Should I add a ROTATE flag to things and pick the nearest one?

**Comment by Kurfursten** [comment]
Sure, I can think of a few ways you could do it.
* Implement it basically like this verb. Simply add two 'arrow keys' one to turn it clockwise and one to turn it counterclockwise
* You could have four buttons instead, one for each cardinal direction, and have the scientist able to simply set preferred direction with one click. Might be a slight timesave when you want to turn it twice.
* Less optimal, but still valid, have it check for an adjacent pile of rubble and face one at random when clicked.

Or were you asking more about coding/implementation? Advice on how to code the interact menu? Help with the .tmpl file if you're updating to modern interface?

**Comment by DamianX** [comment]
Oh, in my mind I had imagined replacing ALL the rotate verbs with a thing like this, like you'd have a rotate button on the HUD, much like the resist button, for instance. I now see what you meant.

**Comment by Kurfursten** [comment]
That's actually not an awful idea, but changing the global HUD seems like it might be something to talk to everyone about with some better visibility.

I mean, just imagine how much more robust the average non-macroing man would become at spinning chairs.

**Pull request merged**